src/main/java/org/checkerframework/specimin/StandardTypeRuleDependencyMap.java

```diff
@@ -314,4 +339,149 @@ public List<Node> getRelevantElements(Object resolved) {
+    Set<MethodDeclaration> alreadyImplemented = new HashSet<>();
+
+    List<ClassOrInterfaceType> parents = new ArrayList<>();
+
+    if (typeDecl instanceof NodeWithExtends<?> withExtends) {
+      parents.addAll(withExtends.getExtendedTypes());
+    }
+    if (typeDecl instanceof NodeWithImplements<?> withImplements) {
+      parents.addAll(withImplements.getImplementedTypes());
+    }
+
+    getAllMustImplementMethodsImpl(parents, typeDecl, typeDecl, methods, alreadyImplemented);
+
+    methods.removeAll(alreadyImplemented);
+    return methods;
+  }
+
+  /**
+   * Helper method for getAllMustImplementMethods. This method recursively finds all must implement
+   * methods in the type declaration's parents and ancestors.
+   *
+   * @param parents The parents of the type declaration, i.e., the extended/implemented types
+   * @param originalTypeDecl The original type declaration
+   * @param typeDecl The current type declaration being processed
+   * @param result The result list
+   * @param alreadyImplemented A set of methods that are already implemented in a superclass or
+   *     interface
+   */
+  private void getAllMustImplementMethodsImpl(
+      List<ClassOrInterfaceType> parents,
+      TypeDeclaration<?> originalTypeDecl,
+      TypeDeclaration<?> typeDecl,
+      List<MethodDeclaration> result,
+      Set<MethodDeclaration> alreadyImplemented) {
+    for (ClassOrInterfaceType type : parents) {
+      try {
+        ResolvedType resolved = type.resolve();
+
+        if (!resolved.isReferenceType()) {
+          continue;
+        }
+
+        boolean isInterface = false;
+        ResolvedReferenceTypeDeclaration resolvedTypeDecl =
+            resolved.asReferenceType().getTypeDeclaration().orElse(null);
+
+        Set<ResolvedMethodDeclaration> methods;
+
+        if (resolvedTypeDecl instanceof ReflectionClassDeclaration reflectionClassDeclaration) {
+          methods = reflectionClassDeclaration.getDeclaredMethods();
+        } else if (resolvedTypeDecl
+            instanceof ReflectionInterfaceDeclaration reflectionInterfaceDeclaration) {
+          isInterface = true;
+          methods = reflectionInterfaceDeclaration.getDeclaredMethods();
+        } else {
+          continue;
+        }
+
+        for (ResolvedMethodDeclaration resolvedMethodDecl : methods) {
+          // Don't preserve equals(Object) and hashCode()
+          if (resolvedMethodDecl.getName().equals("equals")
+              && resolvedMethodDecl.getNumberOfParams() == 1
+              && resolvedMethodDecl.getParam(0).getType().describe().equals("java.lang.Object")) {
+            continue;
+          }
+
+          if (resolvedMethodDecl.getName().equals("hashCode")
+              && resolvedMethodDecl.getNumberOfParams() == 0) {
+            continue;
+          }
+
+          for (MethodDeclaration methodDecl : originalTypeDecl.findAll(MethodDeclaration.class)) {
+            if (areAstAndResolvedMethodLikelyEqual(resolvedMethodDecl, methodDecl)) {
```

Comment:
Why can't we use FQNs for the parameters in `areAstAndResolvedMethodLikelyEqual`? That would make this correct beyond a doubt.